pkg/virt-handler/cmd-client/client.go

```diff
@@ -260,8 +260,16 @@ func FindPodDirOnHost(vmi *v1.VirtualMachineInstance) (string, error) {
 	return "", fmt.Errorf("No command socketdir for vmi %s", vmi.UID)
 }
 
-// gets the cmd socket for a VMI
-func FindSocketOnHost(vmi *v1.VirtualMachineInstance) (string, error) {
+// SocketFinder represents an interface for finding sockets on the host.
+type SocketFinder interface {
+	FindSocketOnHost(vmi *v1.VirtualMachineInstance) (string, error)
+}
```

Comment:
This is still very low level interface. Is there any common functionality that uses this function?

Reply:
@fossedihelm Could you please have a look?

Reply:
@anishbista60 how can i implement the high level interface ? Could you please suggest me and i'll proccedd as per the requirement ?